pjdfstest.c

```diff
@@ -545,10 +547,20 @@ show_stat(struct stat64 *sp, const char *what)
 		printf("%lld", (long long)sp->st_blocks);
 	else if (strcmp(what, "atime") == 0)
 		printf("%lld", (long long)sp->st_atime);
-	else if (strcmp(what, "mtime") == 0)
-		printf("%lld", (long long)sp->st_mtime);
+	else if (strcmp(what, "atime_ns") == 0)
+		printf("%lld", (long long)sp->st_atim.tv_nsec);
 	else if (strcmp(what, "ctime") == 0)
 		printf("%lld", (long long)sp->st_ctime);
+	else if (strcmp(what, "ctime_ns") == 0)
+		printf("%lld", (long long)sp->st_ctim.tv_nsec);
+	else if (strcmp(what, "mtime") == 0)
+		printf("%lld", (long long)sp->st_mtime);
+	else if (strcmp(what, "mtime_ns") == 0)
+		printf("%lld", (long long)sp->st_mtim.tv_nsec);
+	else if (strcmp(what, "birthtime") == 0)
```

Comment:
A #ifdef will need to be added for this: FreeBSD is the only OS that supports `.birthtime` in `struct stat`.